src/HeapedVectors.jl

```diff
@@ -0,0 +1,156 @@
+__precompile__()
+
+struct HeapedVector{T, F<:Function}  
```

Comment:
I think `HeapVector` is a better name.

Reply:
Is this really that important, I mean `HeapedVector` is also giving an intuitive idea of whats going on in there , so isnt it fine to go give this name only .